src/stylesheets/components/_header.scss

```diff
@@ -1,98 +1,118 @@
-// Header variables ---------- //
+// Header helpers
+// ---------------------------------
+
+@mixin nav-border-extended {
+  @include add-bar(
+    0.5,
+    'primary',
+    'bottom',
+    0,
+    2
+  );
+}
 
-$height-nav-secondary: units(6);
-$width-nav-sidebar:    units('card-lg');
+@mixin nav-border-basic {
+  @include add-bar(
+    0.5,
+    'primary',
+    'bottom',
+    0,
+    2,
+    -0.5
+  );
+}
 
-$z-index-header:  z-index(300);
-$z-index-overlay: z-index(400);
-$z-index-nav:     z-index(500);
+$z-index-header:  300;
+$z-index-overlay: 400;
 
-// Header ------------- //
+// Header
+// ---------------------------------
 
 .usa-header {
   @include typeset($theme-font-family-header);
-  width: 100%;
-  z-index: $z-index-header;
-
-  @include at-media($theme-nav-width) {
-    border-bottom: units(1px) solid color('base-lighter');
-  }
+  @include clearfix;
+  z-index: z-index($z-index-header);
 
   a {
     border-bottom: none;
   }
 
+  // The search <form>
   .usa-search {
     @include at-media($theme-nav-width) {
       float: right;
     }
+  }
 
-    [role=search] {
-      @include at-media($theme-nav-width) {
-        float: right;
-        max-width: calc(#{$search-min-width} + #{$usa-btn-small-width});
-        width: inherit;
-      }
+  // Accessibility: The <div> with search role
+  [role=search] {
+    @include at-media($theme-nav-width) {
+      float: right;
+      max-width: calc(#{$search-min-width} + #{$usa-btn-small-width});
+      width: 100%;
     }
   }
+
+  // The search <input>
+  [type=search] {
+    min-width: 0; // Fix a Firefox display quirk
+  }
+
+  + .usa-hero {
+    border-top: units(1px) solid color('white');
+  }
+
+  + .usa-section,
+  + main {
+    border-top: units(1px) solid color('base-lighter');
```

Comment:
@thisisdano can you help me understand why we're adding the border to the top of the next `.usa-section` or `main` rather than the bottom of the header itself. On the surface, it seems more fragile so just wanted to understand.